packages/vm/src/wasm_backend/store.rs

```diff
@@ -0,0 +1,44 @@
+#[cfg(feature = "cranelift")]
+use wasmer::Cranelift;
+#[cfg(not(feature = "cranelift"))]
+use wasmer::Singlepass;
+use wasmer::{Bytes, Engine, Pages, Store, Target, Tunables as ReferenceTunables, JIT};
+use wasmer_engine::Tunables; // See https://github.com/wasmerio/wasmer/issues/1872
+
+use crate::size::Size;
+
+use super::limiting_tunables::LimitingTunables;
+
+/// Created a store with the default compiler and the given memory limit (in pages)
+pub fn make_store(memory_limit: Size) -> Store {
+    #[cfg(feature = "cranelift")]
+    {
+        let compiler = Cranelift::default();
+        let engine = JIT::new(compiler).engine();
+        make_store_with_engine(&engine, memory_limit)
+    }
+
+    #[cfg(not(feature = "cranelift"))]
+    {
+        let compiler = Singlepass::default();
+        let engine = JIT::new(compiler).engine();
+        make_store_with_engine(&engine, memory_limit)
+    }
+}
+
+/// Created a store with no compiler and the given memory limit (in pages)
+pub fn make_store_headless(memory_limit: Size) -> Store {
```

Comment:
Ah, headless can run pre-compiled code, but not compile itself, which makes it lighter to start up when running many instances, right?

Reply:
Yes. This is also usefull for cross compilation, e.g. for IoT stuff. You compile one machine and run in a much more limited environment.